gunicorn/workers/ggevent.py

```diff
@@ -41,7 +41,7 @@ def patch(self):
         sockets = []
         for s in self.sockets:
             sockets.append(socket.socket(s.FAMILY, socket.SOCK_STREAM,
-                                         fileno=s.sock.fileno()))
+                                         fileno=s.sock.detach()))
```

Comment:
why this change?

Reply:
In this code, the original objects in self.sockets (before line 45) are eventually garbage collected and thus closed, leading to "Bad file descriptor". If the file descriptor gets reused, it can lead to "socket operation on non-socket". If you call socket.detach as in this patch - the socket is preserved by the process and it doesn't matter if socket gets garbage collected.

Reply:
kind reminder @benoitc

Reply:
another kind reminder @benoitc :pleading_face: